tests/ModelAggregateTest.php

```diff
@@ -0,0 +1,318 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Atk4\Data\Tests;
+
+use Atk4\Data\Model\Scope;
+use Atk4\Data\Model\Scope\Condition;
+
+class ModelAggregateTest extends \Atk4\Schema\PhpunitTestCase
+{
+    /** @var array */
+    private $init_db =
+        [
+            'client' => [
+                ['name' => 'Vinny'],
+                ['name' => 'Zoe'],
+            ],
+            'invoice' => [
+                ['client_id' => 1, 'name' => 'chair purchase', 'amount' => 4.0],
+                ['client_id' => 1, 'name' => 'table purchase', 'amount' => 15.0],
+                ['client_id' => 2, 'name' => 'chair purchase', 'amount' => 4.0],
+            ],
+            'payment' => [
+                ['client_id' => 1, 'name' => 'prepay', 'amount' => 10.0],
+                ['client_id' => 2, 'name' => 'full pay', 'amount' => 4.0],
+            ],
+        ];
+
+    /** @var Model\Invoice|null */
+    protected $invoice;
+    /** @var \Atk4\Data\Model|null */
+    protected $invoiceAggregate;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->setDB($this->init_db);
+
+        $this->invoice = new Model\Invoice($this->db);
```

Comment:
move to `createInvoice` method, then we do not need clone too

Reply:
Done